server/container/guice/guice-common/src/main/java/org/apache/james/modules/CommonServicesModule.java

```diff
@@ -65,6 +75,37 @@ protected void configure() {
         bind(serverProbe).in(Singleton.class);
     }
 
+    private void listenToConfigurables() {
+        Configurables configurables = new Configurables();
+        bind(Configurables.class).toInstance(configurables);
+
+        bindListener(new SubclassesOf(Configurable.class), new TypeListener() {
```

Comment:
The anonymous TypeListener deserves a real type.
You should extract the whole "listen to these class instantiation order" into a module or even a module factory.